Panic in sidecar in 1.22.0

**What happened**:

We now got this panic a few times, but I can't say what triggers it:

```
2022-05-04 16:48:15	
{"ctlConf":{"Address":"localhost","IsLocal":false,"LocalFile":"","Delay":0,"Timeout":0,"Test":"","TestSdkName":"","GRPCPort":9357,"HTTPPort":9358},"featureGates":"CustomFasSyncInterval=false\u0026Example=true\u0026NodeExternalDNS=true\u0026PlayerAllocationFilter=true\u0026PlayerTracking=true\u0026SDKGracefulTermination=true\u0026StateAllocationFilter=true","message":"Starting sdk sidecar","severity":"info","source":"main","time":"2022-05-04T14:48:15.111106903Z","version":"1.22.0"}
2022-05-04 16:48:15	
{"gsKey":"ourgame/ourgame-ncnfh-chvkh","message":"Created GameServer sidecar","severity":"info","source":"*sdkserver.SDKServer","time":"2022-05-04T14:48:15.112166489Z"}
2022-05-04 16:48:15	
{"grpcEndpoint":"localhost:9357","message":"Starting SDKServer grpc service...","severity":"info","source":"main","time":"2022-05-04T14:48:15.112648367Z"}
2022-05-04 16:48:15	
{"httpEndpoint":"localhost:9358","message":"Starting SDKServer grpc-gateway...","severity":"info","source":"main","time":"2022-05-04T14:48:15.113900376Z"}
2022-05-04 16:48:15	
panic: runtime error: invalid memory address or nil pointer dereference
2022-05-04 16:48:15	
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1550dc5]
2022-05-04 16:48:15	
2022-05-04 16:48:15	
goroutine 87 [running]:
2022-05-04 16:48:15	
agones.dev/agones/pkg/sdkserver.(*SDKServer).WatchGameServer(0xc00008b040, 0x182abe0, {0x1bb69c0, 0xc000904210})
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/pkg/sdkserver/sdkserver.go:536 +0x1e5
2022-05-04 16:48:15	
agones.dev/agones/pkg/sdk._SDK_WatchGameServer_Handler({0x18f51e0, 0xc00008b040}, {0x1bb3090, 0xc0002a0840})
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/pkg/sdk/sdk.pb.go:904 +0xd0
2022-05-04 16:48:15	
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00035cd80, {0x1bc2488, 0xc000678000}, 0xc000906300, 0xc000625860, 0x28d93a0, 0x0)
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/vendor/google.golang.org/grpc/server.go:1237 +0xd63
2022-05-04 16:48:15	
google.golang.org/grpc.(*Server).handleStream(0xc00035cd80, {0x1bc2488, 0xc000678000}, 0xc000906300, 0x0)
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/vendor/google.golang.org/grpc/server.go:1317 +0xa10
2022-05-04 16:48:15	
google.golang.org/grpc.(*Server).serveStreams.func1.1()
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/vendor/google.golang.org/grpc/server.go:722 +0x98
2022-05-04 16:48:15	
created by google.golang.org/grpc.(*Server).serveStreams.func1
2022-05-04 16:48:15	
	/go/src/agones.dev/agones/vendor/google.golang.org/grpc/server.go:720 +0xef
```

**What you expected to happen**:
no panic

**How to reproduce it (as minimally and precisely as possible)**:
I don't know

**Anything else we need to know?**:
Agones never tried to kill the pod although the sidecar was dead. Is that expected?

**Environment**:
- Agones version: 1.22.0
- Kubernetes version (use `kubectl version`): `Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.5-eks-bc4871b", GitCommit:"5236faf39f1b7a7dabea8df12726f25608131aa9", GitTreeState:"clean", BuildDate:"2021-10-29T23:32:16Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}`
- Cloud provider or hardware configuration: AWS EKS
- Install method (yaml/helm): Helm
- Troubleshooting guide log(s):
- Others:

--- a/pkg/sdkserver/sdkserver.go
+++ b/pkg/sdkserver/sdkserver.go
@@ -199,6 +199,9 @@ func (s *SDKServer) Run(ctx context.Context) error {
 	if !cache.WaitForCacheSync(ctx.Done(), s.gameServerSynced) {
 		return errors.New("failed to wait for caches to sync")
 	}
+
+	// need this for streaming gRPC commands
+	s.ctx = ctx
 	// we have the gameserver details now
 	s.gsWaitForSync.Done()
 
@@ -207,8 +210,6 @@ func (s *SDKServer) Run(ctx context.Context) error {
 		return err
 	}
 
-	// need this for streaming gRPC commands
-	s.ctx = ctx
 	logLevel := agonesv1.SdkServerLogLevelInfo
 	// grab configuration details
 	if gs.Spec.SdkServer.LogLevel != "" {

--- a/pkg/sdkserver/sdkserver.go
+++ b/pkg/sdkserver/sdkserver.go
@@ -207,6 +207,8 @@ func (s *SDKServer) Run(ctx context.Context) error {
 		return err
 	}
 
+	// need this for streaming gRPC commands
+	s.ctx = ctx
 	logLevel := agonesv1.SdkServerLogLevelInfo
 	// grab configuration details
 	if gs.Spec.SdkServer.LogLevel != "" {
@@ -262,8 +264,6 @@ func (s *SDKServer) Run(ctx context.Context) error {
 	}()
 	defer s.server.Close() // nolint: errcheck
 
-	// need this for streaming gRPC commands
-	s.ctx = ctx
 	s.workerqueue.Run(ctx, 1)
 	return nil
 }

